addons/stock_account/models/product.py

```diff
@@ -106,8 +132,23 @@ class ProductProduct(models.Model):
         help="Technical field to correctly show the currently selected company's currency that corresponds "
              "to the totaled value of the product's valuation layers")
     stock_valuation_layer_ids = fields.One2many('stock.valuation.layer', 'product_id')
-    valuation = fields.Selection(related="categ_id.property_valuation", readonly=True)
-    cost_method = fields.Selection(related="categ_id.property_cost_method", readonly=True)
+    cost_method = fields.Selection(string="Cost Method", selection=PRODUCT_COST_METHOD, compute="_compute_cost_method", readonly=True)
+    valuation = fields.Selection(string="Valuation", selection=PRODUCT_VALUATION_SELECTIONS, compute="_compute_valuation", readonly=True)
+
+    @api.depends('categ_id')
+    def _compute_valuation(self):
+        for record in self:
+            if record.categ_id:
+                record.valuation = record.categ_id.property_valuation
+            elif self.env.user.has_group('stock_account.group_stock_accounting_automatic'):
+                record.valuation = 'real_time'
```

Comment:
This is user dependent computation so maybe need to add depend_context and can be move outside loop